panel/widgets/stats_plot_card.py

```diff
@@ -0,0 +1,61 @@
+"""The StatsPlotCard enables the user to display a Dashboard KPI Card with
+
+- Title: The name or a short title
+- Value: Primary Value to display. For example an absolute value.
+- Value2: A secondary value. For example a percentage change.
+- Plot: A plot. One of line", "step", "area", "bar
+
+The card can be layout out as
+
+- a column (text and plot on top of each other) or
+- a row (text and plot after each other)
+
+The card enables performant streaming.
+The text section is responsive and resizes on window resize.
+"""
+import param
+from bokeh.models.sources import ColumnDataSource
+from panel.models.stats_plot_card import StatsPlotCard as _BkStatsPlotCard
+from panel.widgets.base import Widget
+
+
+class StatsPlotCard(Widget):
+    """The StatsPlotCard enables the user to display a Dashboard KPI Card with
+
+    - Title: The name or a short title
+    - Value: Primary Value to display. For example an absolute value.
+    - Value2: A secondary value. For example a percentage change.
+    - Plot: A plot. One of line", "step", "area", "bar
+
+    The card can be layout out as
+
+    - a column (text and plot on top of each other) or
+    - a row (text and plot after each other)
+
+    The card enables performant streaming.
+    The text section is responsive and resizes on window resize.
+    """
+    _widget_type = _BkStatsPlotCard
+
+    _rename = {}
+
+    title = param.String(doc="""The title or a short description of the card""")
+    layout = param.ObjectSelector(default="column", objects=["column"])
+    value = param.String(doc="""The primary value to be displayed""")
+    value2 = param.String(doc="""A secondary value to display together with the primary value""")
+    plot_data = param.ClassSelector(
```

Comment:
Would prefer to support dictionaries of arrays and dataframes here.

Reply:
I sort of do as well. But I would like to support efficient patching and streaming. That is the main goal.  Do you have an existing API or implementation i could clone @philippjfr ?

Reply:
Could try to reuse what we implemented (I refactored your implementation) on the `BaseTable` class.

Reply:
The main reason why using Bokeh models directly is not a good idea is that they generally should only be rendered once and you can end up with odd behavior if you reuse them across multiple views.

Reply:
I have started looking at this. But it leaves me with a lot of questions

1. If I cannot instantiate one `ColumnDataSource` and use it to `stream` and `patch` data, then I don't understand how I can efficiently `stream` and `patch` data in Panel. The efficiency of the ColumnDataSource is essential for streaming (and in general). Resending the whole series every time it updates is too slow and requires too much cpu.
2. I've looked at the `BaseTable` Class. But it contains a lot of functionality specific to tables. It would not be relevant for the `StatsPlotCard`, `PerspectiveViewer` widget or any other widget where you just want to transfer tabular data efficiently. Should I/ someone refactor the essential parts into a `BaseData` or `BaseTabularData` widget?

Reply:
Yes, I'll factor out a TabularData Mixin of some kind. Will look at this asap.

Reply:
Okay, this is done now and I've rewritten the component on top of the ReactiveData baseclass.